Fix search webapp ui

rahulsayini opened the pull request:
add a if check to rdatatable.js and flag isVisible: false to webAppConfig.js

rahulsayini committed:
fix UI of webapp search

rahulsayini committed:
fix UI of webapp search

rahulsayini committed:
add an if condition to rdatatable

rahulsayini committed:
add an if condition to rdatatable

jeddcuison reviewed (changes requested):
I wanted the change to be confined in RDataTable so that if in the future some other part of Jwala uses it and passes out-of-spec data, RDataTable can handle it and you won't have to fix the other parts of Jwala. Please change RDataTable with this code:

    if (tmpVal && (col.isVisible === undefined || col.isVisible)) {
        var tmpStr = tmpVal.toString();
        ...           
    }

In addition, please write a Selenium test case for this. The test case should do the following:

1. Add at least 2 web applications
2. Do a search so much so that the 1st web app is the only one shown
3. Do a search so much so that the 2nd web app is the only one shown
4. Do a search so that 2 of the web apps are shown
5. Do a search in which the result would be no web app displayed
6. Delete the web app you added - cleanup

Note: When creating the web application, follow the way a web app is created in this test - AppCreateTest. JwalaChromeTestSuite is the test that you should run. Change test.properties user and password to n9sf user-password. Do not check in test.properties.

rahulsayini committed:
address code review

rahulsayini committed:
address code review

jeddcuison commented on jwala-webapp/src/main/webapp/resources/js/react/WebAppConfig.js:
```diff
@@ -20,7 +20,7 @@ var WebAppConfig = React.createClass({
                                tableIndex="id.id"
                                colDefinitions={[{key: "id", isVisible: false},
-                                                {key: "secure", renderCallback: this.renderSecureColumn, sortable: false},
+                                                {key: "secure", renderCallback: this.renderSecureColumn, sortable: false, isVisible: false},
```
Can you revert this ? Because we don't need this change anymore. Remember one of the reasons for the change request is to limit the code change in RDataTable.js ? To revert it checkout from master again to a new branch e.g. fix-search-webapp-ui-2, copy the change you did for RDataTable.js, push then create another pull request. Afterwards close this pull request putting in the comment that you have a new pull request. Put the number of the new pull request in the said comment.

jeddcuison reviewed (changes requested):
Please revert the change to WebAppConfig.js since we don't need to change that.

rahulsayini committed:
remove isVisible: false from WebAppConfig.js

rahulsayini committed:
remove isVisible: false from WebAppConfig.js

jmahmoodcc reviewed (changes requested):
Jedd requested a selenium test be written as well. Please create the test that Jedd described.

rahulsayini committed:
Add selenium test cases for webapp search

rahulsayini committed:
Add selenium test cases for webapp search

jeddcuison committed:
Implement tomcat and apache webserver media (#95)

* Write a generic method that can distribute any media

* List all media

* Add "user friendly" name property

* Incorporate Tomcat media to Jvm and refactor media related code

In addition define Apache HTTPD media as part of web server

Code Change Details

- Use ModelMapper to handle DTO-entity mapping
- Refactor Media DTO to mirror JpaMedia class variable types
- Mark distributeWebServer as deprecated
- Transfer path serializer and deserializer to commons
- Update tests affected by the changes
- Refactor UI, rename tomcat variable names and define React keys to options element
- Format some code
- Remove unused import(s)
- Update java doc
- Remove equals and hashCode overrides in Media

* Make Tomcat required and fix bug wherein JDK is  not validated

In addition rewrite comparators to set selected media from "==" to "==="

* Fix media entity to DTO mapping problems

The following is not being mapped

- MediaType
- localPath
- remoteDir

* when deploying the JVM use the tomcat media configured for the JVM

* move MediaType to common

* Implement apache httpd media for web server

Details

- UI drop down for apache httpd
- Creation of new web server with apache httpd
- Update of new web server with apache httpd
- Refactor: write new create and update web server methods that returns the web server entity

* remove remote.tomcat.dir.name from vars.properties

* remove jwala.tomcat.zip.name from vars.properties

* remote remote.tomcat.service.exe from vars.properties

* fix install jvm script to strip dash from apache tomcat version

* allow nullable apacheHttpMediaId when creating a web server from the REST API

* fix for web server not associating the apache httpd media on creation

* Write a generic method that can distribute any media

* List all media

* Add "user friendly" name property

* Incorporate Tomcat media to Jvm and refactor media related code

In addition define Apache HTTPD media as part of web server

Code Change Details

- Use ModelMapper to handle DTO-entity mapping
- Refactor Media DTO to mirror JpaMedia class variable types
- Mark distributeWebServer as deprecated
- Transfer path serializer and deserializer to commons
- Update tests affected by the changes
- Refactor UI, rename tomcat variable names and define React keys to options element
- Format some code
- Remove unused import(s)
- Update java doc
- Remove equals and hashCode overrides in Media

* Make Tomcat required and fix bug wherein JDK is  not validated

In addition rewrite comparators to set selected media from "==" to "==="

* Fix media entity to DTO mapping problems

The following is not being mapped

- MediaType
- localPath
- remoteDir

* when deploying the JVM use the tomcat media configured for the JVM

* move MediaType to common

* Implement apache httpd media for web server

Details

- UI drop down for apache httpd
- Creation of new web server with apache httpd
- Update of new web server with apache httpd
- Refactor: write new create and update web server methods that returns the web server entity

* remove remote.tomcat.dir.name from vars.properties

* remove jwala.tomcat.zip.name from vars.properties

* remote remote.tomcat.service.exe from vars.properties

* fix install jvm script to strip dash from apache tomcat version

* allow nullable apacheHttpMediaId when creating a web server from the REST API

* fix for web server not associating the apache httpd media on creation

* Fix create and update web server tests

* Apply write lock to distributeMedia and remove deprecated methods and properties

* Remove duplicate mock definitions caused by conflict resolution

* Fix failing test

* Do not change the state

Details:

- Fixes bug wherein a new web server's state is changed to STOPPED when it is edited and then saved

* Show MediaType display name instead of the enum name

* Fix NPE

* Replace apache httpd dir with the one from media that is stored in db

* Construct httpd conf paths from values taken from the media table instead of from a property file

* Fix failing tests

* Make generated status path's protocol "http" and ping resource to "index.html"

* Get port from db

* Refactor code

- set busy icon outside of validation method since validation should only do validation
- use lastly to hide the busy icon and re-enable the dialog box
- wrap long line

* Prevent users from doing something with the application until upload is complete

* fix web server not being associated with group after being created and remove deploy dir property

* generate the web server and JVM with a media remote dir specified as a property, and update the postman tests to download the JDK media, Tomcat media, and Apache HTTPD media

* Fix bug wherein file upload fails for IE

The reason why upload was failing is because IE gives the file's name with its full path which when concatenated with the repository path becomes an invalid path

* fix slash missing after the remote.jwala.data.dir property, and remove some hard coded values

* revert changes to allow templatizing the remote dir of media - don't allow templatized remote dir path, and do some clean up for associating the web servers with a group

* Create the directory including any necessary but nonexistent parent directories

In addition, show the whole directory path when there is a failure in creating the directory to make troubleshooting easier

* in postman tests: move delete media test to the end of the test suite and add the tomcat media ID to a JVM creation test

* fix updating webserver when changing groups

* Fix domain user account for create windows service issue

LinHungWu committed:
updating buildVersion for JWALA to 0.0.214 for release

jmahmoodcc commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,64 @@
+        Thread.sleep(1000);
+        driver.findElement(By.xpath("//span[contains(text(), 'Search')]/following-sibling::input")).click();
+        driver.switchTo().activeElement().sendKeys("zApp1");
+        Thread.sleep(1000);
```
Where do you validate that the search results contain the expected app?

jmahmoodcc commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,64 @@
+        driver.switchTo().activeElement().sendKeys("zApp1");
+        Thread.sleep(1000);
+        driver.findElement(By.xpath("//span[contains(text(), 'Search')]/following-sibling::input")).clear();
+        driver.switchTo().activeElement().sendKeys("zApp2");
```
Same thing: where is the validation of the search results?

jmahmoodcc commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,64 @@
+        driver.switchTo().activeElement().sendKeys("zApp2");
+        driver.findElement(By.xpath("//span[contains(text(), 'Search')]/following-sibling::input")).clear();
+        Thread.sleep(1000);
+        driver.switchTo().activeElement().sendKeys("zApp");
```
Again, where is the validation of the search results?

jmahmoodcc reviewed (changes requested):
Maybe I missed, but I didn't see where the search result is validated in the selenium test

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.215-SNAPSHOT

rahulsayini committed:
update httpdconf.tpl and add tear down operations for media

LinHungWu committed:
updating buildVersion for JWALA to 0.0.215 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.216-SNAPSHOT

rahulsayini committed:
update httpdconf.tpl for linux

rahulsayini committed:
Merge branch 'master' of https://github.com/cerner/jwala

LinHungWu committed:
updating buildVersion for JWALA to 0.0.216 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.217-SNAPSHOT

jeddcuison committed:
Modify tests to select tomcat for JVM and select HTTPD for the web server

In addition fix assertion failure on the web server status path due to a code change

jmahmoodcc committed:
Merge pull request #103 from cerner/handle-tomcat-and-httpd-media-selection-in-selenium-tests

Modify tests to select tomcat for JVM and select HTTPD for the web se…

rahulsayini committed:
update httpd conf metadata

jeddcuison committed:
Fix intermittent resource tab bug wherein add resource is not clickable after clicking "Ext Properties" (#102)

jmahmoodcc committed:
update JvmStateReceiverAdapter to be backwards compatible with older ReportingLifeCycleListener (#101)

* update JvmStateReceiverAdapter to be backwards compatible with older version of ReportingLifeCycleListener

* fixes for code review: check for null, fix error messages, and add unit test

* use contains() to check the message key instead of specifically checking the type of the key

* update the comments

LinHungWu committed:
updating buildVersion for JWALA to 0.0.217 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.218-SNAPSHOT

akinny committed:
Update README.md

jmahmoodcc committed:
update the JDK media name in the postman test, not the media dir

jmahmoodcc committed:
fix postman server.xml template to use hostname

rahulsayini committed:
fix test cases

jmahmoodcc committed:
Merge remote-tracking branch 'origin/postman-52-53' into upgrade-1.3.7-to-0.0.217

rahulsayini committed:
Merge pull request #106 from cerner/postman-52-53

fix test cases

jmahmoodcc committed:
change postman JVM template to use property for jgroups cluster name

rahulsayini committed:
take context name from command line

jmahmoodcc committed:
clean up JDK media after updating the name

jmahmoodcc committed:
Merge pull request #107 from cerner/upgrade-1.3.7-to-0.0.217

Upgrade 1.3.7 to 0.0.217

PeterTeunissen committed:
Fix for SOARCOM-2825: running a JVM's service under the NT AUTHORITY\NETWORK SERVICE account should not require a password.

LinHungWu committed:
Merge pull request #108 from cerner/SOARCOM-2825

Fix for SOARCOM-2825: running a JVM's service under the NT AUTHORITY\…

LinHungWu committed:
updating buildVersion for JWALA to 0.0.218 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.219-SNAPSHOT

Sharvaribarve committed:
History logging of resources at web app (#79)

* Added history logging statement

* Getting a single group and changed logging message

* Added history write statements to group and application REST impl

* Removed history write from Application service impl

* Changed Test files to include Config class

* Shifted history logging to ApplicationServiceImpl and logged history for every host in groupserviceRestImpl

* Removed blank lines

* Removed unused imports and changed message

Sharvaribarve committed:
Error Message on clicking show details should display detailed error message instead of [object object] (#96)

* Iterating over content object to display individual properties

* Formmated code

* Added line break

* Reverted changes and added to check if content is an array

* Reverted changes in Resource  and removed extra content check in error alert

* Add test case for deployment of an invalid resource

Check if the error dialog box that is displayed after trying to deploy an invalid resource has the details section (the details section should not be shown)

rahulsayini committed:
address code review comments and add assertions  for selenium

rahulsayini committed:
Merge branch 'fix-search-webapp-ui' of https://github.com/cerner/jwala into fix-search-webapp-ui

# Conflicts:
#	jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/JwalaChromeTestSuite.java
#	jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java

rahulsayini commented:
opened a new branch UI-webapp-search https://github.com/cerner/jwala/pull/111
